packages/chopsticks/src/blockchain/index.ts

```diff
@@ -108,6 +113,44 @@ export class Blockchain {
     return this.#txpool
   }
 
+  async #saveBlockToDB(block: Block) {
+    if (this.db) {
+      const { hash, number, header, extrinsics } = block
+      // delete old ones with the same block number if any, keep the latest one
+      await this.db.getRepository(BlockEntity).delete({ number })
+      await this.db.getRepository(BlockEntity).upsert(
+        {
+          hash,
+          number,
+          header,
+          extrinsics: await extrinsics,
+          parentHash: (await block.parentBlock)?.hash,
+          storageDiff: await block.storageDiff(),
+        },
+        ['hash'],
+      )
+    }
+  }
+
+  /**
+   * Try to get block from db, if pass in number, get block by number, else get block by hash
+   */
+  async #getBlockFromDB(key: number | HexString): Promise<Block | undefined> {
```

Comment:
either make this `loadBlockFromDB` or remove `registerBlock` so a get function doesn't have side effects

Reply:
sure, let's rename it `loadBlockFromDB`